### PR TITLE
Fix regression caused by #2214: incorrect editing state in reading list detail view

### PR DESCRIPTION
**Pull request by nambatee**

### CODE REVIEW

File: Wikipedia/Code/ReadingListDetailViewController.swift [hunk 1]
```diff
@@ -157,7 +157,7 @@ class ReadingListDetailViewController: ColumnarCollectionViewController, Editabl
     // MARK: - Empty state
     
     override func isEmptyDidChange() {
-        guard readingList.countOfEntries == 0 else {
+        guard isEmpty == (readingList.countOfEntries == 0) else {
```
Comment (montehurd) [inline Wikipedia/Code/ReadingListDetailViewController.swift]: Just curious, why `==` vs `||`?

Comment (nambatee) [reply Wikipedia/Code/ReadingListDetailViewController.swift]: you can try https://phabricator.wikimedia.org/T192165 with `||`, also if you swipe to delete in the detail view, then tap `Edit`, the action pane won't open

Comment (nambatee) [reply Wikipedia/Code/ReadingListDetailViewController.swift]: if the collection view goes from an empty state to a nonempty state, `isEmpty || readingList.countOfEntries == 0` (`false || false`)  will be false and we'll never update `editController's` `isCollectionViewEmpty`

Comment (montehurd) [reply Wikipedia/Code/ReadingListDetailViewController.swift]: Oh interesting thank you! I wonder if we should put the contents of that guard into a `let` with a name which conveys what you just explained. If you can think of something that'd be cool. No stress if not.

Comment (montehurd) [reply Wikipedia/Code/ReadingListDetailViewController.swift]: Oh I noticed one more strange related edge case where the empty view doesn't appear:

4 mov

Follow-on is fine if you prefer. Let me know :)

Comment (nambatee) [reply Wikipedia/Code/ReadingListDetailViewController.swift]: Yeah, good point! Let me add that let and fix that bug in this PR :) thanks for checking it out!

**Pull request merged**